extensions/cache/deployment/src/test/java/io/quarkus/cache/test/runtime/ProgrammaticApiTest.java

```diff
@@ -203,6 +214,47 @@ public void testExceptionInValueLoader() throws Exception {
         assertGetIfPresentMissingKey(key);
     }
 
+    @Test
+    public void testCompute() {
+        try {
+            String value = cacheCompute.compute(KEY_COMPUTE, (key, previousValue) -> {
+                assertEquals(KEY_COMPUTE, key);
+                assertNull(previousValue);
+                return VALUE_COMPUTE;
+            });
+            assertEquals(VALUE_COMPUTE, value);
+            assertEquals(VALUE_COMPUTE, cacheCompute.get(KEY_COMPUTE, Function.identity()).await().atMost(Duration.ofSeconds(1)));
+        } finally {
+            cache.invalidate(KEY_COMPUTE);
+        }
+    }
+
+    @Test
+    public void testComputeConcurrent() {
+
+        String CONCURRENT_VALUE = "CONCURRENT_VALUE";
+
+        try {
+            assertThrows(RuntimeException.class, () -> cacheCompute.compute(KEY_COMPUTE, (key, previousValue) -> {
```

Comment:
I cannot understand why this test is failing. 
after a concurrent insert, putIfAbsent still return null and not the concurrently inserted value.
@ben-manes maybe you have an idea about this one ?

Reply:
Where is the CI log with a test failure? I am not familiar with this code base so would need the test rewritten against Caffeine's apis to debug if a contract violation.

The `cacheCompute` and `cache` are two different cache instances, so I don't see how `cacheCompute` is supposed to get the value that is set in `cache`. That value is never returned to the outer cache, so the test does not make sense to me.

Reply:
I don't think the CI has run yet, it is pending validation, but it is failing on my local machine, let me show you the code

Reply:
Please re-read the second paragraph. These are two different cache instances with no data passed between them, yet the test asserts as if it was the same instance. That seems like part of the problem

Reply:
@ben-manes this was the cause, thanks a lot